src/IssueViz/Editor/LanguageDetection/SonarLanguageRecognizer.cs

```diff
@@ -93,5 +100,8 @@ private IEnumerable<IContentType> GetExtensionContentTypes(string fileExtension,
 
         private static bool IsCFamilyDocument(IEnumerable<IContentType> contentTypes) =>
             contentTypes.Any(type => type.IsOfType("C/C++"));
+
+        private static bool IsRoslynDocument(IEnumerable<IContentType> contentTypes) =>
+            contentTypes.Any(type => type.IsOfType("Roslyn Languages"));
```

Comment:
Based on https://github.com/dotnet/roslyn/blob/14b7626db71faee5df16a4db4b339046bf54c95f/src/EditorFeatures/Text/ContentTypeNames.cs#L11

Works for .cs files and .vb files - can you think of anything else I should check? I'm not sure what other files do we analyze.

Reply:
C, C++, C#, VB.NET and JavaScript currently. So I don't think there's anything else we need to cover at the moment.